From ce0587d2216dd3a6feb9713ef6577ae0a341d6f0 Mon Sep 17 00:00:00 2001
From: Jing Hua <jinghua@marvell.com>
Date: Fri, 5 Jun 2015 11:04:50 +0800
Subject: [PATCH 0193/1240] fix: arlp: seperate armada-lp.dts into
 armada-lp.dtsi and armada-lp-db.dts

	Seperate armada-lp.dts into armada-lp.dtsi and armada-lp-db.dts.
	- armada-lp.dtsi is for SoC, has all nodes with detail information but "disabled".
	- armada-lp-db.dts is for board, which include armada-lp.dtsi and "enable" all
	  necessary nodes for this specific board.

Change-Id: I56f9311d5e04f58856196cffe3b9895ba7505e28
Signed-off-by: Jing Hua <jinghua@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/19842
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 arch/arm/dts/Makefile            |  2 +-
 arch/arm/dts/armada-lp-db.dts    | 17 +++++++++++++++++
 arch/arm/dts/armada-lp.dts       | 29 -----------------------------
 arch/arm/dts/armada-lp.dtsi      | 27 +++++++++++++++++++++++++++
 configs/mvebu_armadalp_defconfig |  2 +-
 5 files changed, 46 insertions(+), 31 deletions(-)
 create mode 100644 arch/arm/dts/armada-lp-db.dts
 delete mode 100644 arch/arm/dts/armada-lp.dts
 create mode 100644 arch/arm/dts/armada-lp.dtsi

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index fe08ae6..9cf7606 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -52,7 +52,7 @@ dtb-$(CONFIG_AM33XX) += am335x-boneblack.dtb
 dtb-$(CONFIG_SOCFPGA) += socfpga_cyclone5_socrates.dtb
 
 dtb-$(CONFIG_MVEBU) += armada-385-db.dtb
-dtb-$(CONFIG_MVEBU) += armada-lp.dtb
+dtb-$(CONFIG_MVEBU) += armada-lp-db.dtb
 dtb-$(CONFIG_MVEBU) += armada-8k.dtb
 
 targets += $(dtb-y)
diff --git a/arch/arm/dts/armada-lp-db.dts b/arch/arm/dts/armada-lp-db.dts
new file mode 100644
index 0000000..31bb042
--- /dev/null
+++ b/arch/arm/dts/armada-lp-db.dts
@@ -0,0 +1,17 @@
+
+/dts-v1/;
+
+#include "armada-lp.dtsi"
+
+/ {
+	model = "Marvell Armada_LP Development Board";
+	compatible = "marvell,armada-lp-db", "marvell,armada-lp";
+
+	soc {
+		internal-regs {
+			serial: serial@12000 {
+				status = "okay";
+			};
+		};
+	};
+};
diff --git a/arch/arm/dts/armada-lp.dts b/arch/arm/dts/armada-lp.dts
deleted file mode 100644
index 925397c..0000000
--- a/arch/arm/dts/armada-lp.dts
+++ /dev/null
@@ -1,29 +0,0 @@
-
-/dts-v1/;
-
-/ {
-	model = "Marvell Armada_LP Development Board";
-	compatible = "marvell,armada-lp-db", "marvell,armada-lp";
-
-	#address-cells = <2>;
-	#size-cells = <2>;
-
-	soc {
-	compatible = "marvell,armadalp-mbus";
-	#address-cells = <1>;
-	#size-cells = <0>;
-
-		internal-regs {
-			compatible = "marvell,internal-regs";
-			#address-cells = <1>;
-			#size-cells = <1>;
-			reg = <0xf1000000 0x4>;
-
-			serial: serial@12000 {
-				compatible = "marvell,armadalp-uart";
-				reg = <0x12000 0x20>;
-				status = "okay";
-			};
-		};
-	};
-};
diff --git a/arch/arm/dts/armada-lp.dtsi b/arch/arm/dts/armada-lp.dtsi
new file mode 100644
index 0000000..b81e88b
--- /dev/null
+++ b/arch/arm/dts/armada-lp.dtsi
@@ -0,0 +1,27 @@
+
+/ {
+	model = "Marvell Armada_LP Development Board";
+	compatible = "marvell,armada-lp";
+
+	#address-cells = <2>;
+	#size-cells = <2>;
+
+	soc {
+	compatible = "marvell,armadalp-mbus";
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+		internal-regs {
+			compatible = "marvell,internal-regs";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			reg = <0xf1000000 0x4>;
+
+			serial: serial@12000 {
+				compatible = "marvell,armadalp-uart";
+				reg = <0x12000 0x20>;
+				status = "disabled";
+			};
+		};
+	};
+};
diff --git a/configs/mvebu_armadalp_defconfig b/configs/mvebu_armadalp_defconfig
index 78497c3..692395f 100644
--- a/configs/mvebu_armadalp_defconfig
+++ b/configs/mvebu_armadalp_defconfig
@@ -10,4 +10,4 @@ CONFIG_CMD_SAVEENV=y
 CONFIG_CMD_MEMORY=y
 +S:CONFIG_OF_CONTROL=y
 +S:CONFIG_OF_EMBED=y
-+S:CONFIG_DEFAULT_DEVICE_TREE="armada-lp"
++S:CONFIG_DEFAULT_DEVICE_TREE="armada-lp-db"
-- 
1.9.1

